hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -498,3 +537,11 @@ func callToArmFunction(source ast.Expr, destination ast.Expr, methodName string)
 
 	return results
 }
+
+func typeRequiresCopying(theType astmodel.Type) bool {
```

Comment:
should we put this on `Type` instead?

Reply:
it would be _so nice_ if you could add interfaces to external types…

Reply:
Yeah, I'll move this. One slightly weird thing is that in general it should return true for ObjectType, but that path will never be exercised by this code because it only checks when the source and destination types are the same, and we'll always generate a different dest type in that case.

Reply:
Oh duh, that's not right - this is really "does this type have reference behaviour?", so for ObjectType it should be false.

Reply:
Man, that's not right either! An ObjectType needs deep-copying if it has any reference types inside it. (Makes me feel like maybe I shouldn't promote this up to Type?)

Reply:
I carried on with this for the various Type implementations because it seemed like it could be useful for other situations, but it's not definable for TypeName without having access to the full set of types. So I'm leaving this as a specific utility function for this code (where the philosophical problem happens to be sidestepped).

Reply:
all good!